jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Request.java

```diff
@@ -421,7 +422,16 @@ private void extractQueryParameters()
             try
             {
                 _queryParameters = new Fields(true);
-                UrlEncoded.decodeTo(query, _queryParameters::add, _queryEncoding);
+
+                if (StandardCharsets.UTF_8.equals(_queryEncoding) || _queryEncoding == null && UrlEncoded.ENCODING.equals(StandardCharsets.UTF_8))
```

Comment:
I don't understand the `UrlEncoded.ENCODING.equals(StandardCharsets.UTF_8)` part.
Isn't that comparing a constant to a constant?

Reply:
It is, but the ENCODING constant is possibly set from the `org.eclipse.jetty.util.UrlEncoding.charset` system property, so it could be non-utf-8. JIT should see this as a constant expression and factor it out, but I could do so manually if that would be less confusing?

Reply:
Ah